universal-application-tool-0.0.1/app/services/export/JsonExporter.java

```diff
@@ -0,0 +1,148 @@
+
+/** Exports all applications for a given program as JSON. */
+public class JsonExporter {
+
+  private final ApplicantService applicantService;
+  private final ProgramRepository programRepository;
+
+  @Inject
+  JsonExporter(ApplicantService applicantService, ProgramRepository programRepository) {
+    this.applicantService = checkNotNull(applicantService);
+    this.programRepository = checkNotNull(programRepository);
+  }
+
+  public String export(ProgramDefinition programDefinition) {
+    DocumentContext jsonApplications = makeEmptyJsonArray();
+    PaginationResult<Application> applicationPaginationResult =
+        programRepository.getApplicationsForAllProgramVersions(
+            programDefinition.id(), PaginationSpec.MAX_PAGE_SIZE_SPEC, Optional.empty());
+
+    for (Application application : applicationPaginationResult.getPageContents()) {
+      CfJsonDocumentContext applicationJson = buildJsonApplication(application, programDefinition);
+      jsonApplications.add("$", applicationJson.getDocumentContext().json());
+    }
+
+    return jsonApplications.jsonString();
+  }
+
+  private CfJsonDocumentContext buildJsonApplication(
+      Application application, ProgramDefinition programDefinition) {
+    ReadOnlyApplicantProgramService roApplicantProgramService =
+        applicantService.getReadOnlyApplicantProgramService(application, programDefinition);
+    CfJsonDocumentContext jsonApplication = new CfJsonDocumentContext(makeEmptyJsonObject());
+
+    jsonApplication.putString(
+        Path.create("program_name"), application.getProgram().getProgramDefinition().adminName());
+    jsonApplication.putLong(Path.create("program_version_id"), application.getProgram().id);
+    jsonApplication.putLong(Path.create("applicant_id"), application.getApplicant().id);
+    jsonApplication.putLong(Path.create("application_id"), application.id);
+    jsonApplication.putString(
+        Path.create("language"),
+        roApplicantProgramService.getApplicantData().preferredLocale().toLanguageTag());
+    jsonApplication.putString(Path.create("create_time"), application.getCreateTime().toString());
+    jsonApplication.putString(
+        Path.create("submitter_email"), application.getSubmitterEmail().orElse("Applicant"));
+
+    if (application.getSubmitTime() == null) {
+      jsonApplication.putNull(Path.create("submit_time"));
+    } else {
+      jsonApplication.putString(Path.create("submit_time"), application.getSubmitTime().toString());
+    }
+
+    for (AnswerData answerData : roApplicantProgramService.getSummaryData()) {
+      // Answers to enumerator questions should not be included because the path is incompatible
+      // with the JSON export schema. This is because enumerators store an identifier value for
+      // each repeated entity, which with the current export logic conflicts with the answers
+      // stored for repeated entities.
+      if (answerData.questionDefinition().isEnumerator()) {
```

Comment:
mmmm this feels like a thing we should sort out? right?

Reply:
we don't provide enumerator question answers to admins anywhere else (i.e. in UI or CSV download), doing it here would be an exception. so far it's not been something they've wanted which is understandable since it's data that's just used to make the questions more readable for the applicant.